test/ParallelIOTest.cpp

```diff
@@ -946,10 +946,16 @@ void hipace_like_write(std::string const &file_ending)
     int const last_step = 100;
     int const my_first_step = i_mpi_rank * int(local_Nz);
     int const all_last_step = last_step + (i_mpi_size - 1) * int(local_Nz);
+
+    bool participate_in_barrier = true;
```

Comment:
@ax3l Can you please check if this bug also affects Hipace? Currently, the sequence of barriers and flushes dont match from rank to rank. This was uncovered only now, since flushing is effectively not collective in many situations, but this test now uses `adios2::Engine::PerformDataWrite()` of BP5 which is a bit stricter there.

Reply:
Ok, great catch!

For HiPACE++, we changed the time stepping logic the other year, so that every MPI rank just writes to exactly one iteration. Thus, it cannot have this bug (anymore).